h/oauth/schemas.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+
+import colander
+
+from h import i18n
+from h.models.auth_client import GrantType, ResponseType
+from h.schemas.base import CSRFSchema
+
+_ = i18n.TranslationString
+
+
+def enum_schema_node(enum_cls, **kwargs):
+    """
+    Return a `colander.SchemaNode` for a field with a given enum type.
+
+    :param enum_cls: The enum class
+    :type enum_cls: enum.Enum
+    """
+    values = []
+    for name, _ in enum_cls.__members__.items():
+        values.append(name)
+
+    return colander.SchemaNode(colander.String(),
+                               validator=colander.OneOf(values),
+                               **kwargs)
+
+
+class CreateAuthClientSchema(CSRFSchema):
+    name = colander.SchemaNode(
+             colander.String(),
+             title=_('Name'),
+             hint=_('This will be displayed to users in the '
+                    'authorization prompt'))
+
+    authority = colander.SchemaNode(
+                  colander.String(),
+                  title=_('Authority'),
+                  hint=_('Set of users whose data this client '
+                         'can interact with'))
+
+    grant_type = enum_schema_node(GrantType,
```

Comment:
We currently support: `GrantType.authorization_code`, `GrantType.jwt_bearer`, `ResponseType.code`.

And as far as I know, we don't have any plans to support any of the other options.